### PR TITLE
[OCPCLOUD-1436] AWS IMDSv2 support

### PR DESCRIPTION
**Pull request by lobziik**
enhancement: https://github.com/openshift/enhancements/pull/1075

Some additional changes in this PR:
- break dependency to cluster-api-aws, due to machine types was moved to openshift/api
- update govmomi due to go 1.16 dependency resolution issues

### CODE REVIEW

**Comment by patrickdillon** [comment]
@lobziik are you hoping to land this for 4.11?

**Comment by lobziik** [comment]
@patrickdillon yes...

File: pkg/types/aws/machinepool.go [hunk 1]
```diff
@@ -25,6 +25,11 @@ type MachinePool struct {
 	// +optional
 	EC2RootVolume `json:"rootVolume"`
 
+	// EC2MetadataOptions defines metadata service interaction options for EC2 instances in the machine pool.
+	//
+	// +optional
+	EC2Metadata `json:"metadataService"`
```
Comment (patrickdillon) [inline pkg/types/aws/machinepool.go:31]: Don't think we should embed this:
```suggestion
	EC2Metadata EC2Metadata `json:"metadataService"`
```

File: pkg/types/aws/validation/machinepool.go [hunk 2]
```diff
@@ -42,6 +44,11 @@ func ValidateMachinePool(platform *aws.Platform, p *aws.MachinePool, fldPath *fi
 	if p.Size < 0 {
 		allErrs = append(allErrs, field.Invalid(fldPath.Child("size"), p.Size, "Storage size must be positive"))
 	}
+
+	if p.EC2Metadata.Authentication != "" && !validMetadataAuthValues.Has(p.EC2Metadata.Authentication) {
+		allErrs = append(allErrs, field.Invalid(fldPath.Child("authentication"), p.EC2Metadata.Authentication, "Metadata authentication must be either Required or Optional"))
```
Comment (patrickdillon) [inline pkg/types/aws/validation/machinepool.go:49]:
```suggestion
		allErrs = append(allErrs, field.Invalid(fldPath.Child("authentication"), p.EC2Metadata.Authentication,  "must be either Required or Optional"))
```
The field name will already be present in the error message

File: hack/tf-fmt.sh [hunk 3]
```diff
@@ -10,7 +10,7 @@ if [ "$IS_CONTAINER" != "" ]; then
 else
   podman run --rm \
     --env IS_CONTAINER=TRUE \
-    --volume "${PWD}:${PWD}:z" \
+    --volume "${PWD}:${PWD}" \
```
[the remark this answers is not in the document]
Comment (lobziik) [reply hack/tf-fmt.sh:13]: Oops, committed that mistakenly. Will remove.
Podman on mac does not support this yet :(

File: data/data/aws/cluster/master/main.tf [hunk 4]
```diff
@@ -146,6 +146,11 @@ resource "aws_instance" "master" {
     ignore_changes = [ami]
   }
 
+  metadata_options {
+    http_endpoint = "enabled"
```
[the remark this answers is not in the document]
Comment (lobziik) [reply data/data/aws/cluster/master/main.tf:150]: I had an issues during installation without this line.

**Comment by lobziik** [comment]
Updated, moved openshift/api update to the first commit along with other vendoring.
There are some code which imports changing just for keep this revision working.

**Comment by patrickdillon** [comment]
/override ci/prow/okd-images

This PR is not breaking okd images, which is having some sort of infrastructure issue ATM.

**Pull request merged**